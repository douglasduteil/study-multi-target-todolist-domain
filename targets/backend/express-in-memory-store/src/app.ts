//

import { json } from "body-parser";
import express from "express";

import listRoute from "./routes/list_todos";

//

export class App {
  public express = express();

  constructor() {
    this.express.use(json());

    this.express.use("/todos", listRoute);
  }
}
