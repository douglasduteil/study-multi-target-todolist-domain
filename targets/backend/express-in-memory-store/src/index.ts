//

import express from "express";

const PORT = 5000;
