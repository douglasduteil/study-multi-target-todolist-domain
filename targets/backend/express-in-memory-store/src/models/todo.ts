//

import { Todo } from "@todolist/core";

//

export type TodoInMemory = Todo;
