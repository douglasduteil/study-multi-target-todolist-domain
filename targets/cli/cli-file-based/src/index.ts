//

import { main } from "./bin";

//

main(process.argv.slice(2));
