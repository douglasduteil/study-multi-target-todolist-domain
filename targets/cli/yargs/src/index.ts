//

import { filterTodosFn } from "@todolist/core";
import { Database } from "@todolist/databases.lowdb";
import { commands, main } from "@todolist/ui.yargs";
import LowdbFileSyncStorageAdapter from "lowdb/adapters/FileSync";
import { version } from "../package.json";

//

const lowDbDataSource = new Database(
  new LowdbFileSyncStorageAdapter("db.json")
);

const { AddCommand, ListCommand } = commands;

main({ version })
  .command(new AddCommand(lowDbDataSource))
  .command(new ListCommand(filterTodosFn, console, lowDbDataSource))
  .parse(process.argv.slice(2));
