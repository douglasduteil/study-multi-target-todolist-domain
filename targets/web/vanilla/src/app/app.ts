//

import { AppComponent } from "@todolist/ui.vanilla";
import { decorate, inject, injectable, postConstruct } from "inversify";
import { IDENTIFIER } from "./identifiers";

//

decorate(injectable(), AppComponent);

//

decorate(
  inject(IDENTIFIER.ROOT_ELEMENT) as ParameterDecorator,
  AppComponent,
  0
);
decorate(inject(IDENTIFIER.DATA_SOURCE) as ParameterDecorator, AppComponent, 1);
decorate(inject(IDENTIFIER.CORE_FILTER) as ParameterDecorator, AppComponent, 2);

//

decorate(postConstruct() as MethodDecorator, AppComponent.prototype, "init");

export class App extends AppComponent {}
