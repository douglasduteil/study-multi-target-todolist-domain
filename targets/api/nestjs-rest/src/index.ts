//

import { NestFactory } from "@nestjs/core";
import { AppModule } from "./app/app.module";

//

async function bootstrap(): Promise<void> {
  const app = await NestFactory.create(AppModule);
  await app.listen(3000);
}

bootstrap();
