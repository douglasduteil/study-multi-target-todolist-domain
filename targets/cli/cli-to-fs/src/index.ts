//

import { Database } from "@todolist.databases/lowdb";
import { commands, main } from "@todolist.ui/yargs";
import { Container, inject, injectable } from "inversify";
import "reflect-metadata";
import { Argv } from "yargs";
import { version } from "../package.json";
import { ListCommand } from "./commands/list";

//

export function cli(adapter?: any): Argv {
  const container = new Container();
  container.bind("DAL").to(DAL);
  container.bind("LOWDB_ADAPTER").toConstantValue(adapter);
 
  container.bind("ListCommand").to(ListCommand);

  return main({ version }).command(container.get("ListCommand"));
}

export function cli2(adapter?: any): Argv {
  const container = new Container();
  container.bind("DAL").to(DAL);
  container.bind("LOWDB_ADAPTER").toConstantValue(adapter);

  const root = main({ version });

  const cmd = commands.list;
  const superHandler = cmd.handler;

  root.command({
    ...cmd,
    handler(args: any): void {
      const options = superHandler(args);

      const dal = container.get<DAL>("DAL");
      console.log(options, dal.todoDB.getTodos());
    }
  });

  return root;
}

//

@injectable()
class DAL {
  public todoDB = new Database(this.adapter);
  constructor(@inject("LOWDB_ADAPTER") private adapter: any) {}
}
