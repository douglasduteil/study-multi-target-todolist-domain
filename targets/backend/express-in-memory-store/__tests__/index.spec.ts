//

import request from "supertest";

import { App } from "src/app";
import { TodoInMemory } from "src/models/todo";

//

const app = new App();

//

test("1 - no todos in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(response.body).toEqual([]);
  }
});
test("2 - add some eggs", async () => {
  {
    // when
    const response = await request(app.express)
      .post("/todos")
      .send({ title: "Buy some eggs" });

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("3 - some eggs in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});
test("4 - add some breed", async () => {
  {
    // when
    const response = await request(app.express)
      .post("/todos")
      .send({ title: "Buy some breed" });

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("5 - some eggs and breed in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});
test("6 - add some milk", async () => {
  {
    // when
    const response = await request(app.express)
      .post("/todos")
      .send({ title: "Buy some milk" });

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("7 - some eggs, breed and milk in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});
test("8 - done buying breed", async () => {
  {
    // given
    const breedTodo = ((await request(app.express).get("/todos"))
      .body as TodoInMemory[]).find(todo => todo.title === "Buy some breed");

    // when
    const response = await request(app.express).patch(
      `/todos/${breedTodo!.id}/completed`
    );

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("9 - some eggs, breed (x) and milk in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});
test("10 - add some chocolate", async () => {
  {
    // when
    const response = await request(app.express)
      .post("/todos")
      .send({ title: "Buy some chocolate" });

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("11 - some eggs, breed (x), milk, chocolate in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});
test("12 - remove chocolate from the list", async () => {
  {
    // given
    const chocolateTodo = ((await request(app.express).get("/todos"))
      .body as TodoInMemory[]).find(
      todo => todo.title === "Buy some chocolate"
    );

    // when
    const response = await request(app.express).delete(
      `/todos/${chocolateTodo!.id}`
    );

    // then
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({ id: expect.any(String) });
  }
});
test("13 - some eggs, breed (x) and milk in the list", async () => {
  {
    // when
    const response = await request(app.express).get("/todos");

    // then
    expect(response.status).toBe(200);
    expect(_(response.body)).toMatchSnapshot();
  }
});

test("foo async await return error no-try-catch", async () => {
  const user = await UserModelfindById();
  if (user instanceof Error) {
    const err = user;
    throw err;
  }

  [err, savedTask] = await to(
    TaskModel({ userId: user.id, name: "Demo Task" })
  );
  if (err) {
    throw new CustomError("Error occurred while saving task");
  }

  if (user.notificationsEnabled) {
    const [err] = await to(
      NotificationService.sendNotification(user.id, "Task Created")
    );
    if (err) {
      console.error("Just log the error and continue flow");
    }
  }

  async function UserModelfindById(): Promise<{ name: string } | Error> {
    return (
      new Promise((resolve, reject) =>
        Math.random() > 0.5
          ? resolve({ name: "gui" })
          : reject("User not found")
      )
        // Alway resolved promise
        .catch(e => e)
    );
  }
});

const _ = (arr: any) =>
  arr.map((val: any) => ({ ...val, id: expect.any(String) }));
