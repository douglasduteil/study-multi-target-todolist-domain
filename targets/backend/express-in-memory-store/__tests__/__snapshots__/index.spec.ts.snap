// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2 - add some eggs 1`] = `
Object {
  "completed": false,
  "id": Any<String>,
  "title": "Buy some eggs",
}
`;

exports[`3 - some eggs in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
]
`;

exports[`4 - add some breed 1`] = `
Object {
  "completed": false,
  "id": Any<String>,
  "title": "Buy some breed",
}
`;

exports[`5 - some eggs and breed in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some breed",
  },
]
`;

exports[`6 - add some milk 1`] = `
Object {
  "completed": false,
  "id": Any<String>,
  "title": "Buy some milk",
}
`;

exports[`7 - some eggs, breed and milk in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some breed",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some milk",
  },
]
`;

exports[`8 - done buying breed 1`] = `
Object {
  "completed": true,
  "id": Any<String>,
  "title": "Buy some breed",
}
`;

exports[`9 - some eggs, breed (x) and milk in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
  Object {
    "completed": true,
    "id": Any<String>,
    "title": "Buy some breed",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some milk",
  },
]
`;

exports[`10 - add some chocolate 1`] = `
Object {
  "completed": false,
  "id": Any<String>,
  "title": "Buy some chocolate",
}
`;

exports[`11 - some eggs, breed (x), milk, chocolate in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
  Object {
    "completed": true,
    "id": Any<String>,
    "title": "Buy some breed",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some milk",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some chocolate",
  },
]
`;

exports[`12 - remove chocolate from the list 1`] = `
Object {
  "completed": false,
  "id": Any<String>,
  "title": "Buy some chocolate",
}
`;

exports[`13 - some eggs, breed (x) and milk in the list 1`] = `
Array [
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some eggs",
  },
  Object {
    "completed": true,
    "id": Any<String>,
    "title": "Buy some breed",
  },
  Object {
    "completed": false,
    "id": Any<String>,
    "title": "Buy some milk",
  },
]
`;
